doc/benchexec-in-container.md

```diff
@@ -0,0 +1,233 @@
+```dockerfile
+FROM debian:bookworm-slim
+# Or whichever base image you are using
+
+# Install python -- add further dependencies of your application here
+RUN apt-get update && apt-get -y install \
+    python3-minimal \
+  && rm -rf /var/lib/apt/lists/*
+
+# TODO Install BenchExec with any method (apt install, pip install, or just copy the .whl)
+# RUN pip install benchexec
+# RUN wget https://github.com/sosy-lab/benchexec/releases/download/<current release>.whl -O /opt/benchexec.whl
+# ... or any other method
+
+# Copy the created script
+COPY init.sh /init.sh
+
+# Set init.sh as the entrypoint -- It is important to use brackets here
+ENTRYPOINT [ "/init.sh" ]
+# Set the default process to run
+CMD [ "bash" ]
+```
+If you already have a Dockerfile, you only need to install BenchExec into it
+and add the last few commands (i.e. copy `init.sh` and set the entrypoint).
+
+With this finished, execute `podman build -t <tag> .` (or
+`docker build -t <tag> .` when using Docker) in the directory where the
+Dockerfile is located to build the container.
+
+### Executing BenchExec in the Container
+
+Start the image with
+```
+  podman run --security-opt unmask=/sys/fs/cgroup --cgroups=split --security-opt unmask=/proc/* --security-opt seccomp=unconfined -it <tag>
+```
+or
+```
+  docker run --privileged --cap-drop=all -it <tag>
+```
+<!--
+In principle, `--security-opt systempaths=unconfined --security-opt seccomp=unconfined`
+should also be sufficient as Docker arguments,
+but then mounting within the container still fails.
+-->
+
+> **IMPORTANT**: The `--privileged` argument, gives the Docker container *full
+> root access* to the host, so make sure to include the `--cap-drop=all` flag,
+> use this command only with trusted images, and configure your Docker
+> container such that everything in it is executed under a different user
+> account, not as root (more details [below](#securing-docker-execution)).
+> BenchExec is not designed to run as root and does not provide any safety
+> guarantees regarding its container under these circumstances.
+
+With this, you should be able to execute, for example, `runexec echo`
+inside the Docker container. (In case you opted for the `.whl` install, you
+need to execute
+`PYTHONPATH=/opt/benchexec.whl python3 -m benchexec.runexecutor <program>`
+instead.)
+
+In case you want to modify this setup, please consider the notes mentioned
+[below](#background-and-technical-details).
+
+### Securing Docker Execution
+
+When running under Docker, the user which executes the commands by default has
+`UID` 0, i.e. has the permissions of `root` on the host system. For example, if
+you mount a folder into the container, the processes started in the container
+have *full write access* to these files, no matter what their permissions are.
+As such, a small mistake in, e.g., an evaluation script that deletes temporary
+files, could easily wreak havoc on your system. Using Podman directly mitigates
```

Comment:
I like that you mentioned a robustness problem such that people do not think "security is not important for me, so I do not need this"!